src/System.Device.Gpio/Interop/Unix/SafeChipHandle.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace System.Device.Gpio
+{
+    internal class SafeChipHandle : SafeHandle
```

Comment:
I'm still not convinced that we want to have all of these pointers from the managed side of things. Same question as before, is it not possible to just pass in ints to the shim and have the shim store these structs?

Reply:
lets say we are not using pointers to chip and pin. In that case OpenPin, SetPinMode methods will not become useful anymore, if we access pins diretly with number without pointer they will be opened, set mode for each read/write, and will be closed back right after, i assume it will be much slower

Reply:
I'm fine either way as long as they're internal